### ISSUE TITLE
`typing.NoReturn` is new in Python 3.5.4 and 3.6.2

### ISSUE THREAD

**Issue by tanbro** [marshmallow-code/webargs#574]
The  `typing.NoReturn` is New in version 3.5.4, 3.6.2

So it can't run on python 3.6.1

**Comment by sirosen** [comment]
Thanks for reporting this!

3.5 isn't supported anymore, but 3.6 is... however, I'm not sure how likely it is that users are on 3.6.1 or 3.6.0?
Looking at [the 3.6 lifecycle](https://www.python.org/dev/peps/pep-0494/), 3.6.2 was in 2017. I would have thought we're in the clear on that front.

@tanbro, may I ask how you ran into this? It will just help me prioritize and figure out if I need to be really focusing on a bugfix release.

I think maybe this could be worked around with something like
```python
try:
    from typing import NoReturn
except ImportError:
    try:
        from typing_extensions import NoReturn
    except ImportError as e:
        raise RuntimeError(
            "python versions not supporting typing.NoReturn "
            "require the use of typing_extensions"
        )
```

if we consider that acceptable (assuming 3.6.1 and 3.6.0 usage is very niche).

**Comment by tanbro** [comment]
Ennn... or just tell webargs users who are using Python 3.6.0, and 3.6.1 to add following code before `import webargs`:

```python
import typing
if not hasattr(typing, 'NoReturn'):
    # typing.NoReturn is new since 3.6.2.
    # It's used in flask-smorest(>=0.26) -> webargs(>=7.0)
    setattr(typing, 'NoReturn', typing.Union[None])

# .....

import webargs
# or 3rd library importing webargs

```


And no changes are needed in webargs itself

**Comment by sirosen** [comment]
@tanbro, given the way you responded, it suggests to me that you're not committed to asking `webargs` to support 3.6.1 , but were just being a good citizen and reporting a bug you noticed. Is that a correct read on the situation?

Given that nobody has shown up to `+1` this, it suggests that there are very few other users seeing this. I also don't see any open issues on `flask-smorest` about this, which is another place where issues might show up.

@lafrech, I'm on the border of treating this as a wontfix. What do you think? We could set `python_requires=">=3.6.2"` in setup.py to draw a line in the sand that we won't try to jump through hoops for 3.6.1 if that appeals to you.

**Comment by lafrech** [comment]
Won't fix is fine with me.

I remember @sloria wontfixing a similar issue in marshmallow. We could require >=3.6.2 but what about bugs that are fixed in a python version that is not the oldest supported major, like 3.7.3? We'd have to require 3.6.2 or >= 3.7.3. Not sure this is even feasible.

OTOH, this is not a bug fixed in 3.6.2 but a feature added there that we're deciding to rely upon, so >=3.6.2 would make it clear indeed.

**Comment by tanbro** [comment]
Thanks, @sirosen @lafrech . The problem is rare, and uses who face it can patch it themselves easily. So, it can be closed
